src/RestControllers/FHIR/FhirExportRestController.php

```diff
@@ -0,0 +1,491 @@
+            return $response;
+        }
+    }
+
+    /**
+     * Return's an HTTP response with the result of the delete operation.  The delete response returns a status
+     * code of StatusCode::ACCEPTED.  Subsequent calls to the DELETE will return a StatusCode::BAD_REQUEST as the
+     * export job has been deleted or marked for deleted.
+     * @param $jobId The unique id of the job.
+     * @return ResponseInterface
+     */
+    public function processDeleteExportForJob($jobId)
+    {
+        // simulate async process
+        // return's 202 that we are starting the process
+        $response = (new Psr17Factory())->createResponse(StatusCode::ACCEPTED);
+        // we could return an OperationOutcome here, but the spec says its optional, so we'll skip it
+
+        // need to grab all of the resource files created and then delete them
+
+
+
+        return $response;
+    }
+
+    /**
+     * Given a job that contains a list of resources to export, process each of those resources and save their results
+     * into Documents stored inside OpenEMR in the ndjson format.  There will be one file per resource and the resources
+     * are saved in the FHIR_DOCUMENT_FOLDER under the file name of <resource>-<jobId>.ndjson with the ndjson
+     * mimetype.  The results and any errors of the export are saved back into the ExportJob.  The ExportJob's status
+     * is marked as completed upon completion.
+     * @see ExportJob::OUTPUT_FORMAT_FHIR_NDJSON
+     * @see FhirExportRestController::FHIR_DOCUMENT_FOLDER
+     * @param ExportJob $job  The job to start processing the resource exports for.
+     * @return ExportJob
+     */
+    public function processResourceExportForJob(ExportJob $job)
+    {
+
+        $processedJob = clone $job;
+        // we will generate a bunch of documents here and then save them out to our folder....
+        $document = new \Document();
+        $folder = self::FHIR_DOCUMENT_FOLDER;
+        $docService = new DocumentService();
+        $categoryId = null;
+        $ouputResult = [];
+        $errorResult = [];
+
+        foreach ($job->getResources() as $resource) {
+            if (!$this->isValidResource($resource)) {
+                $errorResult[] = $this->createOperationOutcomeError("Resource does not support export operation");
+                continue;
+            }
+
+            $fileName = $resource . "-" . $job->getId() . ".ndjson";
+            $fullPath = $job->getId() . DIRECTORY_SEPARATOR . $fileName;
+
+            // TODO: @adunsulag this might work for small exports... but we may need to change this if we start dealing
+            // with a ton of data...
+            $lines = [];
+            for ($i = 0; $i < 5; $i++) {
+                // for now we are just testing with dummy data until we actually run through each resource.
+                $patient = new FHIRPatient();
+                $fhirId = new FHIRId();
+                $fhirId->setValue(Uuid::uuid4());
```

Comment:
is this uuid the job id or something else? (just trying to figure out best way to incorporate openemr's uuid scheme here)

Reply:
Fixed this as it was just being used for testing.